lib/stripe/stripe_object.rb

```diff
@@ -72,13 +72,17 @@ def refresh_from(values, opts, partial=false)
     #
     # * +values+ - Hash of values to use to update the current attributes of
     #   the object.
+    # * +:opts+ Options for StripeObject like an API key.
     #
     # ==== Options
     #
-    # * +:opts+ Options for StripeObject like an API key.
-    def update_attributes(values, opts = {})
+    # * +:dirty+ Whether values should be initiated as "dirty" (unsaved).
+    #   Defaults to true.
+    def update_attributes(values, opts = nil, options = {})
+      dirty = options[:dirty]
       values.each do |k, v|
-        @values[k] = Util.convert_to_stripe_object(v, opts)
+        @values[k] = Util.convert_to_stripe_object(v, opts || {})
+        dirty_value!(@values[k]) unless dirty == false
```

Comment:
@mmcgrana Great point here. I've converted the pretty ugly `dirty == false` over to a `#fetch` use in dd11f4b. Again, optional arguments will make this code significantly more clear once we can use them them, so I've added a TODO on the subject.